opencog/atomspace/ImportanceIndex.cc

```diff
@@ -123,3 +146,35 @@ UnorderedHandleSet ImportanceIndex::getHandleSet(
 	               [](Atom* atom)->Handle { return atom->getHandle(); });
 	return ret;
 }
+
+UnorderedHandleSet ImportanceIndex::getMaxBinContents()
+{
+    UnorderedHandleSet ret;
+    for (AtomSet s : boost::adaptors::reverse(idx))
+    {
+        if (s.size() > 0)
+        {
+	        std::transform(s.begin(), s.end(), inserter(ret),
+	               [](Atom* atom)->Handle { return atom->getHandle(); });
+	        return ret;
+        }
+    }
+
+    return ret;
+}
+
+UnorderedHandleSet ImportanceIndex::getMinBinContents()
+{
+    UnorderedHandleSet ret;
+    for (AtomSet s : idx)
```

Comment:
What I want to get is the smallest/biggest bin that is not empty. So iterate over all the bins starting from the bottom/top and if I find a bin that is not empty I immediately return that. Maybe you missed the return at line 159.